ChaosMod/Effects/db/Peds/PedsIntervention.cpp

```diff
@@ -0,0 +1,87 @@
+#include <stdafx.h>
+
+//Effect by ProfessorBiddle
+
+
+//future plans: make jesus spawn above and parachute to player, possibly jesus sniper in helicopter
+static void OnStart()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	Vector3 playerPos = GET_ENTITY_COORDS(playerPed, true);
+	static constexpr Hash modelHash = -835930287;
+
+	Hash relationshipGroup;
+	ADD_RELATIONSHIP_GROUP("_DIVINE_JESUS", &relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(0, relationshipGroup, GET_HASH_KEY("PLAYER"));
+	SET_RELATIONSHIP_BETWEEN_GROUPS(0, GET_HASH_KEY("PLAYER"), relationshipGroup);
+	for (int i = 0; i < 4; i++)
```

Comment:
Could this not be accomplished as a single loop of 20, _or_ with a nested for loop (outer being 4, inner being 5)? Given that a lot of things are repetitive, feels like you could just be calling `SET_PED_FIRING_PATTERN` based off a Hash array.

Something like...
```cpp
Hash patterns[] = {0xC6EE6B4C, ...};

for (int i = 0; i < 20; i++) {
  Ped ped = CreatePoolPed(...);
  
  SET_PED_SUFFERS_CRITICAL_HITS(ped, false);
  ...
  SET_PED_FIRING_PATTERN(ped, pattern[i % 5]);
}
```

I'm not up to date on my C++, but. Thinking out loud.

Reply:
I was trying to do sets of 5 each with different weapons and firing patterns which would be hard in a loop, but it could probably be done in a loop of 20 yeah

Reply:
If you nested the loops though, could you not achieve the same thing in less code?

```cpp
/* Group */
for (int i = 0; i < 4; i++) {
  /* Peds */
  for (int j = 0; j < 5; j++) { 
    Ped ped = CreatePoolPed(...);
    ...
  }
}
```

Since at least right now, you have 4 identical groups, no? So 5 peds with `WEAPON_RAYPISTOL` and firing pattern `0xC6EE6B4C` and so on and so forth